### DISCUSSION TITLE
Cara menyisipkan icon di x.vote?

### DISCUSSION THREAD

**Discussion by yuliakbard** [mecha-cms/mecha#283] [Q&A] [answered]
ane menyimpan icon seperti ini kang

``` php
<?= self::get('components/icon',['path' => $state->y->linoo->icon->{'thumbs_up'},'size' => 20]); ?>
```

sedangkan untuk menampilkan vote nya seperti ini:

``` php
  <?= self::form('vote', [
      'key' => 'y',
      'title' => 'Yes <span role="status">%d</span>'
  ]); ?>
```

Output yang ane harapkan itu seperti:

``` txt
👍 120
```

bagaimana cara menyisipkan nya kang ke dalam value 'title'?

Kalau ane bungkus di luar nya

seperti:

``` php
<div class="votes">
  <?= self::get('components/icon',['path' => $state->y->linoo->icon->{'thumbs_up'},'size' => 22]); ?>
  <?= self::form('vote', [
      'key' => 'y',
      'title' => 'Yes <span role="status">%d</span>'
  ]); ?>
</div>
```

untuk triggernya tetap pada teks `Yes <span role="status">%d</span>`.

**Comment by taufik-nurrohman** [comment, marked as answer]
~~~ php
<div class="votes">
  <?php $icon = self::components('icon', [
      'path' => $state->y->linoo->icon->{'thumbs_up'},
      'size' => 22
  ]); ?>
  <?= self::form('vote', [
      'key' => 'y',
      'title' => $icon . ' <span role="status">%d</span>'
  ]); ?>
</div>
~~~

**Reply by yuliakbard** [reply]
owh iya, mesti tambahin titik ya kang. hihihi.

`self::components()` ini manggil isi dalam folder kek gini bisa juga ya kang

**Reply by taufik-nurrohman** [reply]
Bisa, sama aja nanti yang ter-eksekusi adalah metode `Layout::__callStatic()`.

**Reply by yuliakbard** [reply]
```
<div class="votes">
  <?php $icon = self::components('icon', [
      'path' => $state->y->linoo->icon->{'thumbs_up'},
      'size' => 22
  ]); ?>
  <?= self::form('vote', [
      'key' => 'y',
      'title' => `'%d' > 0 ? $icon . ' <span role="status">%d</span>' : $icon` //ini gak bisa dikondisionalkan ya kang
  ]); ?>
</div>
```

**Reply by taufik-nurrohman** [reply]
Itu `` ` `` di kanan-kiri buat apa mas?

Jangan pakai `'%d' > 0` tapi `$page->votes['y'] > 0`:

~~~ php
<?php $icon = '👍'; ?>
<?php $vote = $page->votes['y'] ?? 0; ?>
<?= self::form('vote', [
    'key' => 'y',
    'title' => $vote > 0 ? $icon . ' <span role="status">' . $vote . '</span>' : $icon
]); ?>
~~~

**Reply by yuliakbard** [reply]
kang kalau ditambahin `key`, saat submit/klik tombol vote terjadi error 403

**Reply by taufik-nurrohman** [reply]
Kemungkinan karena update ekstensi Layout. Besok pagi tak coba benerin. Coba down-grade ekstensi Layout ke versi 2.0.0 dulu.

**Reply by taufik-nurrohman** [reply]
Coba ini mas https://github.com/mecha-cms/x.layout/commit/fc6762aab4941303d134e9fda8e6bbc5ab3adcc1

**Reply by yuliakbard** [reply]
masih belum bisa kang, masih error 403 Forbidden

**Reply by taufik-nurrohman** [reply]
Ada konflik dengan ekstensi lain mungkin. Dari saya aman soalnya. Mas ada berkas log _error_ terakhir?

Kalau ada rute kustom coba dinonaktifkan semua dulu, atau buat prioritasnya jadi lebih tinggi dari rute ekstensi Vote.

**Reply by yuliakbard** [reply]
sudah coba nonaktifkan semua ekstensi kang, sudah tested di instalan mecha yg basic requirement (mecha,x.user,x.panel,x.vote), masih sama kang masalahnya. apa ada pengaruh settingan htaccess/ permission file kang

**Reply by yuliakbard** [reply]
ini ane tested online, gk ada masalah kang `https://roku.lokersumatera.com/loker/1`

**Reply by taufik-nurrohman** [reply]
Masalahnya cuma di lokal aja berarti ya.

**Reply by taufik-nurrohman** [reply]
https://stackoverflow.com/a/43141944/1163000

**Reply by yuliakbard** [reply]
untuk saat ini problem di local nya saja kang, cuma gk tau nih kalau di online masih berjalan normal di hostingan sekarang dan mungkin kalau beda beda panel hostingan nanti ngaruh kek gimananya. dulu kan pernah sempat masalah kek gini juga. fixed sendiri karena ganti layanan hostingan yg beda panel

**Reply by taufik-nurrohman** [reply]
Cek link mas, itu bug spesifik ke Windows.

**Reply by taufik-nurrohman** [reply]
https://github.com/mecha-cms/x.vote/releases/tag/v1.0.1

**Reply by yuliakbard** [reply]
fixed kang  😍😍
